projects/plugins/jetpack/extensions/shared/use-autosave-and-redirect/index.js

```diff
@@ -0,0 +1,91 @@
+/**
+ * External dependencies
+ */
+import { noop } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { useSelect, dispatch } from '@wordpress/data';
+import { useState } from '@wordpress/element';
+
+function redirect( url, callback ) {
+	if ( callback ) {
+		callback( url );
+	}
+	window.top.location.href = url;
+}
+
+export default function useAutosaveAndRedirect( redirectUrl, onRedirect = noop ) {
+	const [ isRedirecting, setIsRedirecting ] = useState( false );
+
+	const { isAutosaveablePost, isDirtyPost, currentPost } = useSelect( select => {
```

Comment:
Unfortunately it looks like using this hook is causing the Recurring Payments JS tests to fail. We might need to add some mocks to the existing tests?

![image](https://user-images.githubusercontent.com/14988353/116637537-858a5480-a9a7-11eb-98bc-7daa86e37a10.png)

Reply:
If it's of any use, I created a hook test in the [ GIF block](https://github.com/Automattic/jetpack/blob/master/projects/plugins/jetpack/extensions/blocks/gif/test/use-fetch-giphy-data.js) for reference